lib/Conversion/TorchToTosa/TorchToTosa.cpp

```diff
@@ -3232,6 +3233,74 @@ LogicalResult ConvertAtenOp<AtenGatherOp>::matchAndRewrite(
   return success();
 }
 
+template <>
+LogicalResult ConvertAtenOp<AtenIndexTensorOp>::matchAndRewrite(
+    AtenIndexTensorOp op, OpAdaptor adaptor,
+    ConversionPatternRewriter &rewriter) const {
+  // Python example of this algorithm:
+  // https://gist.github.com/AmosLewis/c90c1148a96291db93408b3fa39f9ae2
+  auto input = adaptor.getSelf();
+  auto inputTensorType =
+      adaptor.getSelf().getType().dyn_cast<RankedTensorType>();
+  // Check input is a tensor type.
+  if (!inputTensorType)
+    return rewriter.notifyMatchFailure(
+        op, "Only tensor types input are currently supported");
+
+  // Deal with torch.prim.ListConstruct of non const value to get the index
+  auto tensorList = op.getIndices();
+  SmallVector<Value> tensorsTorchType;
+  if (!getListConstructElements(tensorList, tensorsTorchType))
+    return op.emitError(
+        "unimplemented: the tensor list is not from list construct");
+  auto tensors = getTypeConvertedValues(rewriter, op->getLoc(),
+                                        getTypeConverter(), tensorsTorchType);
+
+  auto index = tensors[0];
```

Comment:
Is there a plan if the size of tensors is > 1? Return a failure with a message noting that it is currently unsupported? Looking at the Torch to linalg version, it appears that multiple index tensors adds quite a bit of complication. I'm not sure how often index lists are used.

Reply:
Add it to todo